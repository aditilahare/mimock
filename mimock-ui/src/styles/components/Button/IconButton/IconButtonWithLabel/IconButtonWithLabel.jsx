import React from 'react';
import PropTypes from 'prop-types';
import { IconButtonData } from '../../ButtonData';
import { IconButtonWithLabelContainer } from './IconButtonWithLabel.style';
import { AiFillCopy, AiOutlineCheck } from 'react-icons/ai';
import { BiTrash, BiPlusMedical } from 'react-icons/bi';
import { FaFilter } from 'react-icons/fa';
import CloseIcon from '../CloseIcon.jsx';

function IconButtonWithLabel({
	label,
	width,
	variant,
	onclickHandler,
	dataTestid,
}) {
	const { color, background } = getVariant(variant);

	return (
		<IconButtonWithLabelContainer
			data-testid={dataTestid}
			$color={color}
			$width={width}
			$background={background}
			onClick={onclickHandler}
		>
			{getIcon(variant)}
			{label}
		</IconButtonWithLabelContainer>
	);
}

function getVariant(variant) {
	return !variant ? IconButtonData.DEFAULT_BUTTON : IconButtonData[variant];
}

function getIcon(variant) {
	switch (variant) {
		case 'CLEAR_BUTTON':
			return <CloseIcon />;
		case 'DELETE_BUTTON':
			return <BiTrash />;
		case 'ADD_BUTTON':
			return <BiPlusMedical />;
		case 'COPY_BUTTON':
			return <AiFillCopy />;
		case 'FILTER_BUTTON':
			return <FaFilter />;
		default:
			return <AiOutlineCheck />;
	}
}

IconButtonWithLabel.propTypes = {
	label: PropTypes.string.isRequired,
	onclickHandler: PropTypes.func,
	dataTestid: PropTypes.string,
	variant: PropTypes.string,
	width: PropTypes.string,
};

export default IconButtonWithLabel;
