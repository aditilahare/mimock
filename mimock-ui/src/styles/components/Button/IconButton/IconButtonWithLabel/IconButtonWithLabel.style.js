import tw from 'tailwind-styled-components';

export const IconButtonWithLabelContainer = tw.button`
  flex
  items-center
  gap-3
  px-9
  py-3
  m-auto
  shadow-md
  rounded-md
  transition-all
  text-center
  w-fit
  text-lg
  font-semibold
  cursor-pointer
  hover:translate-y-0.5
  hover:brightness-105
	${(props) => Object.values(props.variant)}
`;
