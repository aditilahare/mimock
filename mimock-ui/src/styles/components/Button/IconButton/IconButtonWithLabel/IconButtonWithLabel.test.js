import React from 'react';
import { render } from '@testing-library/react';
import IconButtonWithLabel from './IconButtonWithLabel';

describe('Icon Button With Label', () => {
	let tree;

	it('should render add Button', async () => {
		tree = await render(
			<IconButtonWithLabel
				dataTestid='icon-with-label-button-test-1'
				background={'bg-green-300'}
				color={'text-lime-700'}
				width='w-5/6'
				label='Add'
				onclickHandler={() => {
					console.log('Clicked add icon button');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('icon-with-label-button-test-1')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render clear Button', async () => {
		tree = await render(
			<IconButtonWithLabel
				dataTestid='icon-with-label-button-test-2'
				background={'bg-red-300'}
				color={'text-red-800'}
				width='w-5/6'
				label='Clear'
				onclickHandler={() => {
					console.log('Clicked clear icon button');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('icon-with-label-button-test-2')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render delete Button', async () => {
		tree = await render(
			<IconButtonWithLabel
				dataTestid='icon-with-label-button-test-3'
				background={'bg-red-600'}
				color={'text-white'}
				width='w-5/6'
				label='Delete'
				onclickHandler={() => {
					console.log('Clicked delete icon button');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('icon-with-label-button-test-3')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render copy Button', async () => {
		tree = await render(
			<IconButtonWithLabel
				dataTestid='icon-with-label-button-test-4'
				background={'bg-blue-600'}
				color={'text-white'}
				width='w-5/6'
				label='Copy'
				onclickHandler={() => {
					console.log('Clicked copy icon button');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('icon-with-label-button-test-4')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render default Button when variant is null', async () => {
		tree = await render(
			<IconButtonWithLabel
				dataTestid='icon-with-label-button-test-5'
				label='Default'
				onclickHandler={() => {
					console.log('Clicked default icon button');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('icon-with-label-button-test-5')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
