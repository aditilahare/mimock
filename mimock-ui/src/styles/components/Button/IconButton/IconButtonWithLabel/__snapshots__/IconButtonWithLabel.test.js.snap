// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button With Label should render add Button 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-black w-fit bg-white dark:bg-slate-300"
    data-testid="icon-with-label-button-test-1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
    Add
  </button>
</div>
`;

exports[`Icon Button With Label should render clear Button 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-black w-fit bg-white dark:bg-slate-300"
    data-testid="icon-with-label-button-test-2"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
    Clear
  </button>
</div>
`;

exports[`Icon Button With Label should render copy Button 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-black w-fit bg-white dark:bg-slate-300"
    data-testid="icon-with-label-button-test-4"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
    Copy
  </button>
</div>
`;

exports[`Icon Button With Label should render default Button when variant is null 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-black w-fit bg-white dark:bg-slate-300"
    data-testid="icon-with-label-button-test-5"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
    Default
  </button>
</div>
`;

exports[`Icon Button With Label should render delete Button 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-black w-fit bg-white dark:bg-slate-300"
    data-testid="icon-with-label-button-test-3"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
    Delete
  </button>
</div>
`;

exports[`Icon Button With Label should render home Button 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-3 px-9 py-3 m-auto shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-white w-fit bg-indigo-600 dark:bg-indigo-800"
    data-testid="icon-with-label-home-button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
      />
    </svg>
    Go back to Dashboard
  </button>
</div>
`;
